backend/src/models/users.js

```diff
@@ -31,7 +31,18 @@ const findById = (id) => {
 
 // Get specific user by email.
 const findByEmail = (email) => {
-  return knex(table).where("email", email);
+  return knex
+    .select(
+      "users.id",
+      "users.name",
+      "users.email",
+      knex.raw(
+        "(SELECT json_build_object('value', COALESCE(users.role_id,0), 'label', COALESCE(roles.display_name,''))) AS role_id"
+      )
+    )
+    .from(table)
+    .leftJoin(rolesTable, { "public.users.role_id": `${rolesTable}.id` })
+    .where("email", email);
```

Comment:
I forgot to add a ticket for this but the email column isn't unique so you can't guarantee only one user will be returned in this query. The fix should just involve adding a migration to make it unique so maybe we can add that now?